src/Compiler/Checking/PatternMatchCompilation.fs

```diff
@@ -194,7 +194,15 @@ let ilFieldToTastConst lit =
     | ILFieldInit.Double f -> Const.Double f
 
 exception CannotRefute
-let RefuteDiscrimSet g m path discrims =
+
+[<Struct>]
+[<RequireQualifiedAccess>]
+type internal CounterExampleType = 
```

Comment:
Nit-picking: the whole module is `internal`, so maybe there's no need to specify it additionally for nested types.